pkg/feature/servicemesh/conditions.go

```diff
@@ -21,6 +22,15 @@ const (
 	duration = 5 * time.Minute
 )
 
+func EnsureAuthNamespaceExists(f *feature.Feature) error {
```

Comment:
so the whole reason to introduce this function instead of calling `feature.CreateNamespaceIfNotExists(...) `in PreCondition is for the `auth.namespace` value to end with `--auth-provider`?

Reply:
I did try to use `feature.CreateNamespaceIfNotExists`. The whole history:

* When looking at other usages (like [here](https://github.com/opendatahub-io/opendatahub-operator/blob/4d8ba7ed4c6183e40ca24eb5dfa4708d853d9ea9/controllers/dscinitialization/servicemesh_setup.go#L63)) I quickly found that you need to bind the namespace value when configuring the feature. So, there is no way to use `feature.CreateNamespaceIfNotExists` directly if you need to do some processing (or maybe there is, but I didn't see it).
* To workaround the previous, I then tried using something like `WithData(ResolveAuthNamespace)` and wrapping the func, similarly to this:
```go
func (feature *Feature) error {
  feature.CreateNamespaceIfNotExists(feature.Spec.field)(feature)
}
```
and I tried to use that in `PreConditions`.
* When trying and testing the previous, I found that `PreConditions` run first and, later, `WithData` runs. This was leading to operating on an empty string because, since `WithData` runs at a later stage, the needed `feature.Spec.field` with the manipulated string is still not set.
* So, I fell back to creating this new func.